lib/Dialect/ESI/ESITypes.cpp

```diff
@@ -0,0 +1,89 @@
+//===- ESITypes.cpp - ESI types code defs -----------------------*- C++ -*-===//
+//
+// Definitions for esi data types. Anything which doesn't have to be public
+// should go in here.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/ESI/ESITypes.hpp"
+#include "circt/AutoGenParser.h"
+#include <llvm/ADT/ArrayRef.h>
+#include <llvm/ADT/Hashing.h>
+#include <llvm/ADT/TypeSwitch.h>
+#include <mlir/IR/Attributes.h>
+#include <mlir/IR/DialectImplementation.h>
+
+using namespace mlir;
+
+namespace circt {
+
+namespace autogen {
+
+// Parser for StringEncoding. Is a specialization of the standard type parsers
+// in AutoGenParser.h.
+template <>
+struct Parse<circt::esi::StringEncoding> {
+  mlir::ParseResult go(MLIRContext *ctxt, DialectAsmParser &parser,
+                       StringRef parameterName,
+                       circt::esi::StringEncoding &result) {
+    StringRef encStr;
+    if (parser.parseKeyword(&encStr))
+      return mlir::failure();
+    if (encStr.compare_lower("ascii") == 0)
```

Comment:
I think that ODS has some support for enums now, can that be used for this?  It may not support "barewords" yet though, that would be a great enhancement.

Reply:
Nice!